modules/carbonAnalyticsAdapter.js

```diff
@@ -0,0 +1,241 @@
+import { logError, generateUUID } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
+import { getStorageManager } from '../src/storageManager.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import adapterManager from '../src/adapterManager.js';
+import adapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+
+const CARBON_GVL_ID = 493;
+const ANALYTICS_VERSION = 'v1.0';
+const PROFILE_ID_KEY = 'carbon_ccuid';
+const PROFILE_ID_COOKIE = 'ccuid';
+const SESSION_ID_COOKIE = 'ccsid';
+const MODULE_NAME = 'carbon';
+const ANALYTICS_TYPE = 'endpoint';
+
+const DAY_MS = 24 * 60 * 60 * 1000;
+const MINUTE_MS = 60 * 1000;
+
+export const storage = getStorageManager({gvlid: CARBON_GVL_ID, moduleName: 'carbon'});
+
+let analyticsHost = '';
+let pageViewId = '';
+let profileId = '';
+let sessionId = '';
+let parentId = '';
+let pageEngagement = {};
+
+let carbonAdapter = Object.assign(adapter({analyticsHost, ANALYTICS_TYPE}), {
+  track({eventType, args}) {
+    args = args ? JSON.parse(JSON.stringify(args)) : {};
+    switch (eventType) {
+      case CONSTANTS.EVENTS.AUCTION_END: {
+        registerEngagement();
+        let event = createBaseEngagementEvent(args);
+        sendEngagementEvent(event, 'auction_end');
+        break;
+      }
+      case CONSTANTS.EVENTS.TCF2_ENFORCEMENT: {
+        registerEngagement();
+        let event = createBaseEngagementEvent(args);
+
+        event.tcf_events = args; // Do we only want relevant information here?
+
+        sendEngagementEvent(event, 'tcf_enforcement');
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+});
+
+// save the base class function
+carbonAdapter.originEnableAnalytics = carbonAdapter.enableAnalytics;
+
+// override enableAnalytics so we can get access to the config passed in from the page
+carbonAdapter.enableAnalytics = function (config) {
+  if (config?.options?.parentId) {
+    parentId = config.options.parentId;
+  } else {
+    logError('required config value "parentId" not provided');
+  }
+
+  if (config?.options.endpoint) {
+    analyticsHost = config.options.endpoint;
+  } else {
+    logError('required config value "endpoint" not provided');
+  }
+
+  pageViewId = generateUUID();
+  profileId = getProfileId();
+  sessionId = getSessionId();
+
+  pageEngagement = { // create the initial page engagement event
+    ttl: 60 * 1, // should be multiple of 60, unit is seconds
```

Comment:
ttl seems to always be seconds.

So we only register engagements for the first 60 seconds after the initial engagement? Is that right?

Seems ttl is not configurable either, so is there a reason to do `60 * 1` here?